### ISSUE TITLE
Times out waiting for bootstap upload (Error code: -20)

### ISSUE THREAD

**Issue by TheSlimShadyy** [checkra1n/BugTracker#1166]
**Tell us about your setup:**
1. What iDevice are you using?
iPhone X Global
2. On what version of iOS is it?
iOS 13.4.1
3. What version of checkra1n are you using?
beta 0.10.1
4. What is your host system (OS version? Hackintosh? VM? etc.)?
10.15.1
5. How are you connecting to the device (USB-A? USB-C? Apple/3rd party cable? Through a USB hub?)?
OEM USB-Lightning cable that comes with iphone

**What are the steps to reproduce the issue?**
1. Run the process normaly on Checkra1n
2. Wait for it to boot
3. Get that error message
...

**What do you expect, and what is happening instead?**

For jailbreak to complete

**Does the issue also occur if you enable Safe Mode?**

Yes

**Any other info, error logs, screenshots, ...?**

**Comment by Chengers** [comment]
I just found a lovely redditor who recommended me to use the unofficial checkra1n 0.10 beta from @Silzee on Twitter https://twitter.com/SilzeeJailbreak/status/1245551447334203392 .  I found this build from this user https://www.reddit.com/r/jailbreak/comments/fxf4yz/news_checkra1n_beta_0101_is_now_available_it/fmv1g06/ in the Checkra1n release megathread. 

Just in case you don't want to run something that an internet stranger found

**Comment by mirzank** [comment]
Exactly the same issue as yours. iphone x. 10.4.1 Except im using MacOs 10.15.4. I receive a error -20. 

Before this i have never had any issues jailbreaking my phone. A few times i received an error -20 in the app but my iphone would still end up being jailbroken.

I just went back and tried a few of the older versions 0.10, 0.99, 0.9, 0.9.9, + 5-6 other versions i have. they all result in the same issue.

Safe mode doesn't help. device just stuck on booting, then checkrain gives an error and device reboots to stock.

I've also tried on 1 other mbp on 10.14, and used 3 different cables. all with same issue.

**Comment by AnanthVivekanand** [comment]
This issue is strangely mainly affecting iPhone X, I also had this issue on a previous checkra1n version.

However, I just jailbroke with iPhone X Global, Checkra1n 0.10.1, iOS 13.4.1 on Catalina 10.15.1

**Comment by mirzank** [comment]
> This issue is strangely mainly affecting iPhone X, I also had this issue on a previous checkra1n version.
> 
> However, I just jailbroke with iPhone X Global, Checkra1n 0.10.1, iOS 13.4.1 on Catalina 10.15.1

So how did you fix the error on the previous version ? Did you just wait for a new checkra1n version ?

**Comment by AnanthVivekanand** [comment]
The -20 error for me actually froze my device on the checkra1n boot screen, not sure what that screen is called. I had to wait for my battery to run out and it booted back into normal iOS. This was back in November, around November 20th on Checkra1n 0.9.2 I think. 

I waited for a new checkra1n version before trying again. I am 70% sure it was checkra1n version 0.9.7 that fixed it for me.

**Comment by blueberrywaffie** [comment]
Happening for me as well, iPad Mini 4, iOS 13.4.1, that has never been jailbroken before. I recently tried to jailbreak my iPad Mini 2, was having the same issue and eventually used 0.9.1 and it worked just fine. About to try 0.9.1 on the iPad Mini 4, will edit back the results. 

Im dumb, 0.9.1 doesn't even let you try 13.4.1, so i'm still stuck 

Mine gets stuck after "KPF: Found task_conversion_eval" is printed to the screen 4 times. Sits for a while and then I get the -20 error on my mac. Wait a bit longer and checkrain prints a bunch of stuff to the screen, then reboots to stock.

**Comment by mirzank** [comment]
> Happening for me as well, iPad Mini 4, iOS 13.4.1, that has never been jailbroken before. I recently tried to jailbreak my iPad Mini 2, was having the same issue and eventually used 0.9.1 and it worked just fine. About to try 0.9.1 on the iPad Mini 4, will edit back the results.
> 
> Im dumb, 0.9.1 doesn't even let you try 13.4.1, so i'm still stuck
> 
> Mine gets stuck after "KPF: Found task_conversion_eval" is printed to the screen 4 times. Sits for a while and then I get the -20 error on my mac. Wait a bit longer and checkrain prints a bunch of stuff to the screen, then reboots to stock.

It seems error -20 is a catch all for multiple errors. You get the kpf text. Mine only goes till booting, then restarts after failing to jb. Your error may warrant opening another ticket

**Comment by blueberrywaffie** [comment]
> > Happening for me as well, iPad Mini 4, iOS 13.4.1, that has never been jailbroken before. I recently tried to jailbreak my iPad Mini 2, was having the same issue and eventually used 0.9.1 and it worked just fine. About to try 0.9.1 on the iPad Mini 4, will edit back the results.
> > Im dumb, 0.9.1 doesn't even let you try 13.4.1, so i'm still stuck
> > Mine gets stuck after "KPF: Found task_conversion_eval" is printed to the screen 4 times. Sits for a while and then I get the -20 error on my mac. Wait a bit longer and checkrain prints a bunch of stuff to the screen, then reboots to stock.
> 
> It seems error -20 is a catch all for multiple errors. You get the kpf text. Mine only goes till booting, then restarts after failing to jb. Your error may warrant opening another ticket

Makes sense, I've created this ticket for anyone experiencing this issue as well 
https://github.com/checkra1n/BugTracker/issues/1180
